File: lessons/compiled_lang/cs/src/f20.loop.cs
// testbox: title="while loop"
using System;

class F20Loop
{
    static void Main()
    {
        string answer = "";
        while (answer != "quit")
        {
            Console.Write("Enter your name (quit to Exit): ");
            answer = Console.ReadLine();

            if (answer != "quit")
                Console.WriteLine($"Hello {answer}!");
        }
    }
}

File: lessons/compiled_lang/java/src/f20.loop.java
// testbox: title="while loop"
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class F20Loop {
    public static void main(String[] args) throws IOException {
        BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
        String answer = "";
        while (!answer.equals("quit")) {
            System.out.print("Enter your name (quit to Exit): ");
            answer = stdin.readLine();

            if (!answer.equals("quit"))
                System.out.println("Hello " + answer + "!");
        }
    }
}
